Utility/Remove-ResourceGroups.ps1

```diff
@@ -0,0 +1,228 @@
+
+.ICONURI 
+
+.EXTERNALMODULEDEPENDENCIES 
+
+.REQUIREDSCRIPTS 
+
+.EXTERNALSCRIPTDEPENDENCIES 
+
+.RELEASENOTES
+
+#>
+<# 
+.SYNOPSIS 
+  Connects to Azure and removes all resource groups which match the name filter
+ 
+.DESCRIPTION 
+  This runbook connects to Azure and removes all resource groups which match the name filter. 
+  You can run across multiple subscriptions, delete all resource groups, or run in preview mode. 
+ 
+  REQUIRED AUTOMATION ASSETS
+    Authorization to targeted Azure subscriptions using one of the following options:
+        Option 1: An Automation credential asset that contains the Azure AD user credential.
+        Option 2: An Automation connection asset that contains the Azure AD service principal.
+
+.PARAMETER AuthenticationType 
+  Mandatory.
+  The type of authentication to use for connection to Azure subscriptions.
+  Valid values are AADCREDENTIAL and SERVICEPRINCIPAL.
+    - AADCREDENTIAL = Automation credential asset using an Azure AD user credential. Uses credential asset name "DefaultAzureCredential" if none specified.
+    - SERVICEPRINCIPAL = Automation connection asset using an Azure AD service principal. More info at https://azure.microsoft.com/documentation/articles/automation-sec-configure-azure-runas-account/
+
+.PARAMETER AuthenticationAssetName 
+  Mandatory. 
+  The name of an authentication asset with authorization for this subscription. 
+ 
+.PARAMETER ActionType 
+  Mandatory. 
+  The specific action to take for either keeping assets that match the name filter and deleting everything else or deleting just those assets that match the name filter. 
+  Valid values are KEEP, DELETE, and DELETEALL.
+    - KEEP = Delete everything except resource groups that match the name filter
+    - DELETE = Delete only resource groups that match the name filter
+    - DELETEALL = Delete all resource groups
+ 
+.PARAMETER SubscriptionIds 
+  Mandatory 
+  Allows you to specify the targeted subscription id(s) for removal of resource groups.   
+  Pass multiple subscripription ids through a comma separated list.   
+ 
+.PARAMETER NameFilter 
+  Optional 
+  Allows you to specify a name filter to limit the resource groups that you will KEEP or DELETE.
+  Pass multiple name filters through a comma separated list.     
+  The filter is not case sensitive and will match any resource group that contains the string.   
+ 
+.PARAMETER PreviewMode 
+  Optional with default of $true. 
+  Execute the runbook to see which resource groups would be deleted but take no action.  
+
+.EXAMPLE
+    Remove-ResourceGroups `
+        -AuthenticationType ServicePrincipal `
+        -AuthenticationAssetName AzureRunAsConnection `
+        -ActionType DELETE `
+        -SubscriptionIds abcd1234-567e-8f9a-1a2b-3c4de5fa6b78,bbcd1234-567e-8f9a-1a2b-3c4de5fa6b78 `
+        -NameFilter removeme,removemetoo
+
+.NOTES
+    AUTHOR: System Center Automation Team
+    LASTEDIT: August 8, 2016
+#> 
+
+workflow Remove-ResourceGroups
```

Comment:
How much faster is the script using For-Each -Parallel?  I think that is the only workflow capability you are using here, right?  Regular PS scripts start much faster since they don't need to be compiled, so I'd recommend making a regular PS Script unless the workflow capabilities are necessary.